pkg/resources/gateways/externalip.go

```diff
@@ -25,22 +25,23 @@ import (
 	"github.com/banzaicloud/istio-operator/pkg/k8sutil"
 )
 
-func (r *Reconciler) GetGatewayAddress() ([]string, error) {
+func (r *Reconciler) GetGatewayAddress() ([]string, bool, error) {
 	var service corev1.Service
 	var ips []string
+	var hasHostname bool
 
 	err := r.Get(context.Background(), types.NamespacedName{
 		Name:      r.gatewayName(),
 		Namespace: r.gw.Namespace,
 	}, &service)
 	if err != nil {
-		return nil, err
+		return nil, false, err
```

Comment:
nit: this could be `hasHostname`, just to keep the pattern.
```suggestion
		return nil, hasHostname, err
```